src/test/scala/is/hail/asm4s/ASM4SSuite.scala

```diff
@@ -420,4 +420,46 @@ class ASM4SSuite extends TestNGSuite {
 
     assert(fb.result()()() == 1)
   }
+
+  @Test def fbFunctionsCanBeNested(): Unit = {
+    val fb = FunctionBuilder.functionBuilder[Boolean]
+    val fb2 = fb.newDependentFunction[Int, Boolean]
+    val localF = fb.newField[AsmFunction1[Int, Boolean]]
+
+    val wrappedInt = Code.invokeStatic[java.lang.Integer, Int, java.lang.Integer]("valueOf", 0)
+    val rawOut = localF.load().invoke[java.lang.Object, java.lang.Object]("apply", wrappedInt)
+
+    fb2.emit(true)
+    fb.emit(Code(
+      fb2.newInstance(localF),
```

Comment:
This seems like a weird interface to me?

Why not have `newInstance` put the allocated function onto the stack and the callee can store it in a class field if they want to?

I guess you'd need the dependent function to have a reference to the parent FB so it can create a local variable for use during initialization. You could also pull that off using `DUP`.